OpenRA.Mods.Common/Widgets/WorldInteractionControllerWidget.cs

```diff
@@ -128,13 +128,17 @@ public override bool HandleMouseInput(MouseInput mi)
 						var unit = World.ScreenMap.ActorsAtMouse(mousePos)
 							.WithHighestSelectionPriority(mousePos, mi.Modifiers);
 
-						if (unit != null && unit.Owner == (World.RenderPlayer ?? World.LocalPlayer))
+						// Players to be included in the selection (the viewer or all players in "Disable shroud" / "All players" mode)
+						var viewer = (World.LocalPlayer == null || World.LocalPlayer.Spectating) ? World.RenderPlayer : World.LocalPlayer;
+						var eligiblePlayers = viewer == null || (viewer.NonCombatant && viewer.Spectating) ? World.Players : new[] { viewer };
```

Comment:
What is the `viewer.NonCombatant` check here for? In general I think this can be rewritten as
```
var viewer = World.RenderPlayer ?? World.LocalPlayer;
var eligiblePlayers = viewer == null || viewer.Spectating ? World.Players : new[] { viewer };
```
(I didn't spot any bugs with that, but maybe I oversaw something.)

Reply:
The `viewer.NonCombatant` check is there to make sure that this is the special "Everyone" player. Not having it causes a bug when `LocalPlayer` is not `null` (the player has surrendered/lost) - things under shroud get selected:
![изображение](https://user-images.githubusercontent.com/1355810/62499960-2b1d8d80-b7ed-11e9-85b0-ccb01758f3e6.png)

I think I managed to find a way to use the standard viewer check. The conditions become a little lengthy so I guess pulling them in variables makes things clearer:
```csharp
var viewer = World.RenderPlayer ?? World.LocalPlayer;
var isShroudDisabled = viewer == null || (World.RenderPlayer == null && World.LocalPlayer.Spectating);
var isEveryone = viewer.NonCombatant && viewer.Spectating;
var eligiblePlayers = isShroudDisabled || isEveryone ? World.Players : new[] { viewer };
```

Reply:
A slight change and I think it works fine now. I didn't spot any bugs.
```csharp
var isEveryone = viewer != null && viewer.NonCombatant && viewer.Spectating;
```